app/controllers/users_controller.rb

```diff
@@ -1,5 +1,8 @@
 class UsersController < ApplicationController
-  def show
+  before_action :require_login, except: [:new, :create]
+
+  def index
+    @users = User.all
```

Comment:
Side note: to be very careful when you call `.all` when you have a large application

Reply:
Is this where you would paginate your view and populate it in batches of say ten?